src/components/InjectedComponents/PostDialog.tsx

```diff
@@ -0,0 +1,247 @@
+                    </Typography>
+                    <SelectRecipientsUI
+                        items={props.availableShareTarget}
+                        selected={props.currentShareTarget}
+                        onSetSelected={props.onSetSelected}
+                        {...props.SelectRecipientsUIProps}
+                    />
+                </DialogContent>
+                <DialogActions className={classes.actions}>
+                    <Button
+                        className={classes.button}
+                        style={{ marginLeft: 'auto' }}
+                        color="primary"
+                        variant="contained"
+                        disabled={props.postBoxButtonDisabled}
+                        onClick={props.onFinishButtonClicked}>
+                        {geti18nString('post_dialog__button')}
+                    </Button>
+                </DialogActions>
+            </ResponsiveDialog>
+        </div>
+    )
+}
+
+export interface PostDialogProps extends Partial<PostDialogUIProps> {
+    textPayloadKey?: keyof I18NStrings
+    imagePayloadKey?: keyof I18NStrings
+    reason?: 'timeline' | 'popup'
+    identities?: Profile[]
+    onRequestPost?: (target: (Profile | Group)[], text: string) => void
+    onRequestReset?: () => void
+}
+export function PostDialog(props: PostDialogProps) {
+    const people = useFriendsList()
+    const groups = useGroupsList()
+    const availableShareTarget = or(
+        props.availableShareTarget,
+        useMemo(() => [...groups, ...people], [people, groups]),
+    )
+    const currentIdentity = or(props.currentIdentity, useCurrentIdentity())
+    const isSteganography = useValueRef(steganographyModeSetting)
+
+    const onRequestPost = or(
+        props.onRequestPost,
+        useCallback(
+            async (target: (Profile | Group)[], text: string) => {
+                const [encrypted, token] = await Services.Crypto.encryptTo(
+                    text,
+                    target.map(x => x.identifier),
+                    currentIdentity!.identifier,
+                )
+                const activeUI = getActivatedUI()
+                if (isSteganography) {
+                    activeUI.taskPasteIntoPostBox(
+                        geti18nString(props.imagePayloadKey ?? 'additional_post_box__steganography_post_pre'),
+                        {
+                            warningText: geti18nString('additional_post_box__encrypted_failed'),
+                            shouldOpenPostDialog: false,
+                        },
+                    )
+                    activeUI.taskUploadToPostBox(encrypted, {
+                        warningText: geti18nString('additional_post_box__steganography_post_failed'),
+                    })
+                } else {
+                    activeUI.taskPasteIntoPostBox(
+                        geti18nString(props.textPayloadKey ?? 'additional_post_box__encrypted_post_pre', encrypted),
```

Comment:
Hi @Jack-Works @SunriseFox Here twitter and facebook use different template in same message record. So may be we should update the message structure in `message.json` like this:

```json
{
    "additional_post_box__encrypted_post_pre": {
        "facebook.com": {
            "message": "Decrypt this post with #Maskbook ! $1",
            "description": "$1 should at the end of the string."
        },
        "twitter.com": {
            "message": "#Maskbook ([I:b])\nDecrypt this tweet with maskbook.com @ProjectMaskbook.\n—§— /* {$1} */"
        }
    }
}
```

What do you think about it?

Reply:
@guanbinrui this format is originally comes from https://developer.chrome.com/extensions/i18n and we may go back to old format so it's better not to change the format other than the chrome one

Reply:
I'll add a commit to this pull request resolve this problem.

Reply:
@guanbinrui with https://github.com/DimensionDev/Maskbook/pull/499/commits/8e8682b9923a3e2a862698d9d9e4c69ceb2e1b81 I think you can resolve this problem now

Reply:
cool~ looks good!

Reply:
Hi @Jack-Works I refactored it with `8e1ca8079f77761defc1ed6c5d5e37fa14e42dfb`. Pls have a look!